src/tribler-core/tribler_core/config/tribler_config.py

```diff
@@ -2,627 +2,152 @@
 Configuration object for the Tribler Core.
 """
 import logging
-import os
 import traceback
-from typing import Optional
+from pathlib import Path
 
 from configobj import ConfigObj, ParseError
 from validate import Validator
 
-from tribler_common.simpledefs import MAX_LIBTORRENT_RATE_LIMIT
+from tribler_core.config.config_registry import SPECIFICATION_REGISTRY
 from tribler_core.exceptions import InvalidConfigException
-from tribler_core.modules.libtorrent.download_config import get_default_dest_dir
-from tribler_core.utilities import path_util
-from tribler_core.utilities.install_dir import get_lib_path
-from tribler_core.utilities.path_util import Path
 
-CONFIG_FILENAME = 'triblerd.conf'
-SPEC_FILENAME = 'tribler_config.spec'
-CONFIG_SPEC_PATH = get_lib_path() / 'config' / SPEC_FILENAME
 
+# fmt: off
 
-class TriblerConfig:
-    """
-    Holds all Tribler Core configurable variables.
-
-    This class is a wrapper around a ConfigObj. It has a specification of it's configuration sections and fields,
-    their allowed values and default value in `config.spec`.
-    """
-
-    def __init__(self, state_dir, config_file=None, reset_config_on_error=False):
-        """
-        Create a new TriblerConfig instance.
 
-        :param config_file: path to existing config file
-        :param reset_config_on_error: Flag indicating whether to recover from corrupt config using default config.
-        :raises an InvalidConfigException if ConfigObj is invalid
-        """
-        self._logger = logging.getLogger(self.__class__.__name__)
-        self._state_dir = Path(state_dir)
+class TriblerConfig:
```

Comment:
The most significant changes are located in this class.
So, it is a good idea to start a review from this particular class.